File: MissionRocketLaunch/MissionTeegardenB/app/src/main/java/com/flame4ost/missionteegardenb/Simulation.java
package com.flame4ost.missionteegardenb;

import android.content.Context;
import android.content.res.AssetManager;


import java.io.IOException;

import java.util.ArrayList;
import java.util.Scanner;

public class Simulation {
    private Context context;

    public Simulation(Context context) {
        this.context = context;
    }

    public ArrayList<Item> downloadPhaseOne() throws IOException {

        AssetManager ass_maneger = context.getAssets();
        ArrayList<Item> loadManifest = new ArrayList();

        Scanner filescanner = new Scanner(ass_maneger.open("Phase-1.txt"));

        while (filescanner.hasNextLine()) {

            Item newItem = new Item();
            String[] tokens = filescanner.nextLine().split("=");
            newItem.name = tokens[0];
            newItem.weight = Integer.parseInt(tokens[tokens.length-1]);

            loadManifest.add(newItem);
        }

        return loadManifest;
    }

    public ArrayList<Item> downloadPhaseTwo() throws IOException {

        AssetManager ass_maneger = context.getAssets();
        ArrayList<Item> loadManifest = new ArrayList();

        Scanner filescanner = new Scanner(ass_maneger.open("Phase-2.txt"));

        while (filescanner.hasNextLine()) {

            Item newItem = new Item();
            String[] tokens = filescanner.nextLine().split("=");
            newItem.name = tokens[0];
            newItem.weight = Integer.parseInt(tokens[tokens.length-1]);

            loadManifest.add(newItem);
        }

        return loadManifest;
    }

    public ArrayList<Rocket> loadU1(ArrayList<Item> list) {
        ArrayList<Rocket> rocket_list_one = new ArrayList<>();
        Rocket rocket_one = new U1();

        for (Item thisItem : list) {
            if (rocket_one.canCarry(thisItem)) {
                rocket_list_one.add(rocket_one);
            } else {
                rocket_one = new U1();
                rocket_list_one.add(rocket_one);
            }
        }
        return rocket_list_one;
    }

    public ArrayList<Rocket> loadU2(ArrayList<Item> list) {

        ArrayList<Rocket> rocket_list_two = new ArrayList();
        Rocket rocket_two = new U2();

        for (Item thisItem : list) {
            if (rocket_two.canCarry(thisItem)) {
                rocket_list_two.add(rocket_two);
            } else {
                rocket_two = new U2();
                rocket_list_two.add(rocket_two);
            }
        }
        return rocket_list_two;
    }

    public int runSimulation(ArrayList<Rocket> rockets) {

        int totalCost = 0;

        for (Rocket currentRocket : rockets) {
            totalCost += currentRocket.getCost();

            while (!currentRocket.launch() || !currentRocket.land()) {
                totalCost += currentRocket.getCost();
            }
        }
        return totalCost;
    }

}

File: MissionRocketLaunch/MissionTeegardenBKotlin/app/src/main/java/com/flame4ost/missionteegardenbkotlin/Simulation.kt
package com.flame4ost.missionteegardenbkotlin

import android.content.Context
import android.content.res.AssetManager
import java.io.IOException
import java.util.*
import kotlin.collections.ArrayList

class Simulation(var context: Context) {

    @Throws(IOException::class)
    fun downloadPhaseOne(): ArrayList<Item>{
        val ass_maneger: AssetManager = context!!.assets
        val loadManifest: ArrayList<Item> = ArrayList()

        val filescanner = Scanner(ass_maneger.open("Phase-1.txt"))

        while (filescanner.hasNextLine()) {
            val newItem: Item = Item()
            val tokens = filescanner.nextLine().split("=").toTypedArray()
            newItem.name = tokens[0]
            newItem.weight = tokens[tokens.size - 1].toInt()
            loadManifest.add(newItem)
        }

        return loadManifest
    }

    @Throws(IOException::class)
    fun downloadPhaseTwo(): ArrayList<Item>{
        val ass_maneger: AssetManager = context!!.assets
        val loadManifest: ArrayList<Item> = ArrayList()

        val filescanner = Scanner(ass_maneger.open("Phase-2.txt"))

        while (filescanner.hasNextLine()) {
            val newItem: Item = Item()
            val tokens = filescanner.nextLine().split("=").toTypedArray()
            newItem.name = tokens[0]
            newItem.weight = tokens[tokens.size - 1].toInt()
            loadManifest.add(newItem)
        }

        return loadManifest
    }

    fun loadU1(list: ArrayList<Item>): ArrayList<Rocket>{
        val rocket_list_one : ArrayList<Rocket> = ArrayList()
        var rocket_one: Rocket = U1()

        for (thisItem in list) {
            if (rocket_one.canCarry(thisItem)) {
                rocket_list_one.add(rocket_one)
            } else {
                rocket_one = U1()
                rocket_list_one.add(rocket_one)
            }
        }
        return rocket_list_one
    }

    fun loadU2(list: ArrayList<Item>): ArrayList<Rocket>{
        val rocket_list_two: ArrayList<Rocket> = ArrayList()
        var rocket_two: Rocket = U2()

        for (thisItem in list) {
            if (rocket_two.canCarry(thisItem)) {
                rocket_list_two.add(rocket_two)
            } else {
                rocket_two = U2()
                rocket_list_two.add(rocket_two)
            }
        }
        return rocket_list_two
    }

    fun runSimulation(rockets: ArrayList<Rocket>): Int{
        var totalCost = 0

        for (currentRocket in rockets) {
            totalCost += currentRocket.getCost()
            while (!currentRocket.launch() || !currentRocket.land()) {
                totalCost += currentRocket.getCost()
            }
        }
        return totalCost
    }
}